### PR TITLE
Respawing team event fix

### PR DESCRIPTION
**Pull request by VALERA771**
Returned wrong value because patch got list before removing values.

Fix for #1757

### CODE REVIEW

**Review by Misaka-ZeroTwo — CHANGES_REQUESTED** [review]
You also need to handle the logic which when ``ev.MaximumRespawnAmount`` got set to a lower value, ``ev.Players`` needs to be adjusted.

**Review by Misaka-ZeroTwo — COMMENTED** [review]
accidentally added a file?

**Comment by VALERA771** [comment]
Oops

**Comment by Misaka-ZeroTwo** [comment]
also it still doesn't solve the problem, because ev.Players only gets adjusted when the eventargs is constructed.

**Comment by VALERA771** [comment]
Okay, now should be fine

**Comment by VALERA771** [comment]
> I feel like it's better to just change the setter in the eventarg instead. cuz that way ev.Players will be the same.

Okay but I don't get what is wrong. ev.Players will be same but list will be adjusted anyway

**Comment by Misaka-ZeroTwo** [comment]
For example
```
ev.MaximumRespawnAmount = 10; // smaller than current ev.Players
List<Player> spawnablePlayers = ev.Players; // this won't work if you are changing afterwards
```

**Comment by VALERA771** [comment]
> For example
> 
> ```
> 
> ev.MaximumRespawnAmount = 10; // smaller than current ev.Players number
> 
> List<Player> spawnablePlayers = ev.Players; // this won't work if you are changing afterwards
> 
> ```

In this case yeah. And it'll be easier

**Comment by VALERA771** [comment]
Tested

**Comment by VALERA771** [comment]
Small thing, I think requires-testing label should be removed

File: Exiled.Events/EventArgs/Server/RespawningTeamEventArgs.cs [hunk 1]
```diff
@@ -58,7 +59,22 @@ public RespawningTeamEventArgs(List<Player> players, int maxRespawn, SpawnableTe
...
         ///     Gets or sets the maximum amount of respawnable players.
         /// </summary>
-        public int MaximumRespawnAmount { get; set; }
+        public int MaximumRespawnAmount
+        {
+            get => maximumRespawnAmount;
+            set
+            {
+                if (value == maximumRespawnAmount)
+                    return;
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Server/RespawningTeamEventArgs.cs:68]: not required

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Pull request merged**